### PR TITLE
[GSoC] Added notebook 204-code-language-id

### PR DESCRIPTION
**Pull request by zilto**
# Changes
I added the notebook 204 for the model `CodeBERTa-language-id` in order to fulfill the pre-requesite of Google Summer of Code 2023.

# Testing
While running CI locally, I only had failures for `build_nbval(macos-11, 3.10)`. At step `Analysing with nbval (MacOS)`, multiple failures are raised and appear to be related to HTTP requests to the Hugging Face Hub.

### CODE REVIEW

**Comment by adrianboguszewski** [comment]
Thank you for your contribution. Before we do the review please do the self-check first (copy the list and tick the item when it's done):
- [ ] each function is described by docstrings and type hints
- [ ] notebook contains explicit descriptions and explanatory diagrams
- [ ] the notebook doesn't use any data (image, video, etc.) that is not CC4.0 licensed 
- [ ] there is a README.md file in consistent style (look at other notebooks)
- [ ] the notebook is added to the main README
- [ ] there are no grammar, punctuation or typo issues (use any free tool for that e.g. Grammarly)
- [ ] there are no committed files besides notebook and readme (please use images or videos from data dir)
- [ ] your PR doesn't change any other notebooks
- [ ] all CI checks passed

**Comment by zilto** [comment]
Hi @adrianboguszewski, here is my checklist!

- [x]  each function is described by docstrings and type hints
- [x]  notebook contains explicit descriptions and explanatory diagrams
- [x]  the notebook doesn't use any data (image, video, etc.) that is not CC4.0 licensed
- [x]  there is a README.md file in consistent style (look at other notebooks)
- [x]  the notebook is added to the main README
- [x]  there are no grammar, punctuation or typo issues (use any free tool for that e.g. Grammarly)
- [ ]  all CI checks passed
^only the `build_docker` check fails. There is no logs to debug and it takes 2h+ to run. How should I go about it?
- [x]  there are no committed files besides notebook and readme (please use images or videos from data dir)
- [x]  your PR doesn't change any other notebooks

**Comment by adrianboguszewski** [comment]
I see. There should be the output for debugging. Let's run it once again.

**Comment by zilto** [comment]
@adrianboguszewski well, the most recent runs produced logs. The `build_docker` check fails for notebook 301, but succeeds for mine (204)

**Comment by adrianboguszewski** [comment]
If you didn't touch 301, don't worry about that :) it will be fixed by us in another PR.

**Comment by zilto** [comment]
Any updates regarding this PR?

**Comment by adrianboguszewski** [comment]
I'll review it this week

File: notebooks/204-code-language-id/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Programming Language Classification with OpenVINO
+
+**Programming language classification** is the task of identifying which programming language is used in an arbitrary code snippet. This can be useful to label new data to include in a dataset, and potentially serve as an intermediary step when input snippets need to be process based on their programming language.
+
+More generally, this tutorial shows how to pull pre-trained models from the Hugging Face Hub using the Hugging Face [Transformers](https://huggingface.co/models) library and convert them to the OpenVINO™ IR format using the [Hugging Face Optimum](https://huggingface.co/docs/optimum) library. You will also learn how to conduct post-training quantization for OpenVINO models using Hugging Face Optimum and do some light benchmarking using the (Hugging Face Evaluate)[https://huggingface.co/docs/evaluate/index] library.
```
Comment (adrianboguszewski) [inline notebooks/204-code-language-id/README.md:5]:
```suggestion
More generally, this tutorial shows how to pull pre-trained models from the Hugging Face Hub using the Hugging Face [Transformers](https://huggingface.co/models) library and convert them to the OpenVINO™ IR format using the [Hugging Face Optimum](https://huggingface.co/docs/optimum) library. You will also learn how to conduct post-training quantization for OpenVINO models using Hugging Face Optimum and do some light benchmarking using the [Hugging Face Evaluate](https://huggingface.co/docs/evaluate/index) library.
```

File: notebooks/204-code-language-id/README.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+This tutorial will be divided in 2 parts:
+1. Create a simple inference pipeline with a pre-trained model using the OpenVINO™ IR format.
+2. Conduct [post-training quantization](https://docs.openvino.ai/latest/ptq_introduction.html) on a pre-trained model using Hugging Face Optimum and benchmark performance.
+
+
+### Licenses
+[`CodeBERTa-small-v1`](https://huggingface.co/huggingface/CodeBERTa-small-v1) and [`CodeBERTa-language-id`](https://huggingface.co/huggingface/CodeBERTa-language-id): no license found on Hugging Face Hub. 
+
+Dataset:
+>Each example in the dataset has is extracted from a GitHub repository, and each repository has its own license. Example-wise license information is not (yet) included in this dataset: you will need to find out yourself which license the code is using.
```
Comment (adrianboguszewski) [inline notebooks/204-code-language-id/README.md:18]:
```suggestion
>Each example in the dataset is extracted from a GitHub repository, and each repository has its own license. Example-wise license information is not (yet) included in this dataset: you will need to find out yourself which license the code is using.
```

**Review by adrianboguszewski — CHANGES_REQUESTED** [review]

**Comment by adrianboguszewski** [comment]
And I tried many times to download models and cannot do it: 
image

**Review by adrianboguszewski — CHANGES_REQUESTED** [review]
This is a great notebook. The only missing part is the inference on the quantized model to prove that it still works as expected. 

And please resolve README file conflict.

**Comment by zilto** [comment]
I added a cell with the "Inference on new input with quantized model", moved from 204 to 244, and fixed the README merge conflict

**Comment by adrianboguszewski** [comment]
Thanks. Approving

**Review by adrianboguszewski — APPROVED** [review]

File: notebooks/244-code-language-id/244-code-language-id.ipynb [hunk 3]
```diff
@@ -0,0 +1,536 @@
+{
```
Comment (eaidova) [inline notebooks/244-code-language-id/244-code-language-id.ipynb:1]: <p>I'm not sure that this image is relevant as there is no convolutions in transformers architecture :) Probably it required some proper description, what it show and which process illustrated</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/974/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

Comment (eaidova) [reply notebooks/244-code-language-id/244-code-language-id.ipynb:1]: <p>I would like to add something "The image bellow demonstrates general post-training quantization flow" before image, there is still not clear that text after image belongs to it as there is no any explicit logical connection between text and image</p><p> </p><p>Or possibly to remove it at all for avoiding confusion</p>

Comment (eaidova) [reply notebooks/244-code-language-id/244-code-language-id.ipynb:1]: @zilto, this the last my small comment, please address, resolve conflicts moving notebook on 245 position, fix typos detected by spellcheck and we are ready to merge it

Comment (zilto) [reply notebooks/244-code-language-id/244-code-language-id.ipynb]: I removed the figure from the notebook and moved the folder and file to 245. For typos:
- Golang: false positive
- Javascript -> JavaScript
- TypeScript: false positive
- superset: false positive
- Husain: false positive
- prerequesites -> prerequisites
- Gb: false positive

I believe everything is in order. Let me know if further changes are necessary

Comment (zilto) [reply notebooks/244-code-language-id/244-code-language-id.ipynb]: How do I add those as part of my PR ? This file doesn't exist on my local version of the repo. Not sure how to proceed

Comment (zilto) [reply notebooks/244-code-language-id/244-code-language-id.ipynb]: @eaidova bump

**Review by eaidova — APPROVED** [review]

**Pull request merged**